test/system/src/test/java/io/pravega/test/system/MultiReaderTxnWriterWithFailoverTest.java

```diff
@@ -72,8 +74,10 @@
     //Duration for which the system test waits for writes/reads to happen post failover.
     //10s (SessionTimeout) + 10s (RebalanceContainers) + 20s (For Container recovery + start) + NetworkDelays
     private static final int WAIT_AFTER_FAILOVER_MILLIS = 40 * 1000;
-    private  List<CompletableFuture<Void>> txnStatusFutureList = new ArrayList<>();
-    private ExecutorService executorService;
+    private final List<EventStreamReader<Long>> readerList = new ArrayList<>();
```

Comment:
These need to be annotated with the lock that guards them. (Or switched for thread safe collections)